SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/ManPages/tsManPagesLibGUI/tsWxPyEventBinder.man

Help on module tsWxPyEventBinder:

NAME
    tsWxPyEventBinder

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxPyEventBinder.py

DESCRIPTION
    tsWxPyEventBinder.py - Instances of this class are used to bind
    specific events to event handlers.

CLASSES
    __builtin__.object
        PyEventBinder
    
    class PyEventBinder(__builtin__.object)
     |  Instances of this class are used to bind specific events to event
     |  handlers.
     |  
     |  Methods defined here:
     |  
     |  Bind(self, target, id1, id2, function, useSystemEventTable=False)
     |      Bind this set of event types to the target
     |  
     |  Unbind(self, target, id1, id2, useSystemEventTable=False)
     |      Remove the event bindings.
     |  
     |  __call__(self, *args)
     |      Backwards compatibility with the old EVT_* functions.
     |      Should be called with either (window, func), (window, ID,
     |      func) or (window, ID1, ID2, func) parameters depending
     |      on the event type.
     |  
     |  __init__(self, evtType, expectedIds=0, evtData=None)
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  EventData
     |      Easy with a property.
     |  
     |  EventType
     |      Easy with a property.
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  eventType
     |      Easy with a property.
     |  
     |  typeId
     |      Easy with a property.

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxPyEventBinder, v1.2.0 (build 04/01/2013)\n\n... ...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxPyEventBinder, v1.2.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxPyEventBinder'
    __version__ = '1.2.0'
    mainTitleVersionDate = 'tsWxPyEventBinder, v1.2.0 (build 04/01/2013)'

VERSION
    1.2.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibGUI/tsWxPkg/src/tsWxPyEventBinder.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:42:27 AM rsg>"
'''
tsWxPyEventBinder.py - Instances of this class are used to bind
specific events to event handlers.
'''
#################################################################
#
# File: tsWxPyEventBinder.py
#
# Purpose:
#
#    Instances of this class are used to bind specific events
#    to event handlers.
#
# Usage (example):
#
#    # Import
#
#    from tsWxPyEventBinder import PyEventBinder
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    2012/06/06 rsg Changed self.evtType to self.ts_EventType.
#                   Added properties eventType and EventType
#                   as aliases for property typeId.
#
#    2012/06/23 rsg Created a placeholder for event instance
#                   argument data such as the nCurses mouse tuple.
#                   Added EventData property and associated
#                   _getEvtData and _SetData methods.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxPyEventBinder'
__version__   = '1.2.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

import tsExceptions as tse

import tsWxGlobals as wx

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

if DEBUG:

    import time

    import tsExceptions as tse
    import tsLogger

#---------------------------------------------------------------------------

class PyEventBinder(object):
    '''
    Instances of this class are used to bind specific events to event
    handlers.
    '''
    def __init__(self, evtType, expectedIds=0, evtData=None):
        '''
        '''
        # Create a placeholder for event instance argument data
        # such as the nCurses mouse tuple.
        self.ts_EventData = evtData

        if DEBUG:
            # Install Logger access
            try:

                # Start the default application logger.
                self.logger = tsLogger.TsLogger(
                    threshold=tsLogger.ERROR,
                    start=time.time(),
                    name=tsLogger.StandardOutputFile)

            except Exception as e:

                self.logger = None
                msg = 'PyEventBinder._init__: Exception = %s' % e
                raise tse.ProgramException(tse.APPLICATION_TRAP, msg)

        # Check expected ids
        if expectedIds not in [0, 1, 2]:
            if DEBUG:
                msg = 'PyEventBinder._init__: Expected ids not 0, 1, or 2'
                self.logger.error(msg)
            raise ValueError(msg)

        self.expectedIds = expectedIds

        # Insure we can iterate on the event type
        self.ts_EventType = wx.tsMakeIterable(evtType)

##        self.thisown = theClass

    #-----------------------------------------------------------------------

    def Bind(self,
             target,
             id1,
             id2,
             function,
             useSystemEventTable=False):
        '''
        Bind this set of event types to the target
        '''
        for et in self.ts_EventType:
            if DEBUG:
                fmt1 = 'tsWxEventBinder.Bind: et=%s; ' % str(et)
                fmt2 = 'target=%s; ' % str(target)
                fmt3 = 'id1=%d; ' % id1
                fmt4 = 'id2=%d; ' % id2
                fmt5 = 'function=%s' % function
                msg = fmt1 + fmt2 + fmt3 + fmt4 + fmt5
                self.logger.debug(msg)
            target.Connect(id1, id2, et, function)

    #-----------------------------------------------------------------------

    def Unbind(self,
               target,
               id1,
               id2,
               useSystemEventTable=False):
        '''
        Remove the event bindings.
        '''
        success = 0
        for et in self.ts_EventType:
            disconnect = target.Disconnect(id1, id2, et)
            if disconnect is not None:
                success += disconnect
            if DEBUG:
                fmt1 = 'tsWxEventBinder.Unbind: et=%s; ' % str(et)
                fmt2 = 'target=%s; ' % str(target)
                fmt3 = 'id1=%d; ' % id1
                fmt4 = 'id2=%d; ' % id2
                fmt5 = 'disconnect=%s' % disconnect
                msg = fmt1 + fmt2 + fmt3 + fmt4 + fmt5
                self.logger.debug(msg)

        return (success != 0)

    #-----------------------------------------------------------------------

    def _getEvtData(self):
        '''
        Easy with a property.
        '''
        return (self.ts_EventData)

    #-----------------------------------------------------------------------

    def _setEvtData(self, evtData):
        '''
        Easy with a property.
        '''
        self.ts_EventData = evtData

    #-----------------------------------------------------------------------

    def _getEvtType(self):
        '''
        Easy with a property.
        '''
        return (wx.tsMakeIterable(self.ts_EventType)) # (self.ts_EventType[0])

    #-----------------------------------------------------------------------

    def __call__(self, *args):
        '''
        Backwards compatibility with the old EVT_* functions.
        Should be called with either (window, func), (window, ID,
        func) or (window, ID1, ID2, func) parameters depending
        on the event type.
        '''
        assert len(args) == 2 + self.expectedIds

        id1 = wx.ID_ANY
        id2 = wx.ID_ANY
        target = args[0]

        if self.expectedIds == 0:

            func = args[1]

        elif self.expectedIds == 1:

            id1 = args[1]
            func = args[2]

        elif self.expectedIds == 2:

            id1 = args[1]
            id2 = args[2]
            func = args[3]

        else:
            if DEBUG:
                msg = 'PyEventBinder.__call__: Unexpected number of IDs'
                self.logger.error(msg)
            raise ValueError(msg)

        self.Bind(target, id1, id2, func)

    #-----------------------------------------------------------------------

    EventData = property(_getEvtData, _setEvtData)
    EventType = property(_getEvtType)
    eventType = property(_getEvtType)
    typeId = property(_getEvtType)

#---------------------------------------------------------------------------

if __name__ == '__main__':

    print(__header__)
